### PR TITLE
Fix PyTorch iterator for multi-gpu, adjust PyTorch example

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
Builds 37773.

**Comment by JanuszL** [comment]
Another build 37779.

**Comment by JanuszL** [comment]
Build after benchmarks fixes 37872 .

File: docs/examples/pytorch/main.py [hunk 1]
```diff
@@ -80,7 +80,9 @@ class HybridTrainPipe(Pipeline):
     def __init__(self, batch_size, num_threads, device_id, data_dir, crop):
         super(HybridTrainPipe, self).__init__(batch_size, num_threads, device_id, seed=12 + device_id)
         self.input = ops.FileReader(file_root=data_dir, shard_id=args.local_rank, num_shards=args.world_size, random_shuffle=True)
-        self.decode = ops.nvJPEGDecoder(device="mixed", output_type=types.RGB)
+        # This paddings set the size for nvJPEG internal buffers to be able to handle all images from full-sized ImageNet
```
Comment (pribalta) [inline docs/examples/pytorch/main.py]: Minor but should be:

This padding sets the size of the internal nvJPEG buffers...

**Pull request merged**